app/controllers/CreateSubscription.scala

```diff
@@ -0,0 +1,127 @@
+package controllers
+
+import actions.CustomActionBuilders
+import actions.CustomActionBuilders.AuthRequest
+import admin.settings.{AllSettingsProvider, SettingsSurrogateKeySyntax}
+import cats.data.EitherT
+import cats.implicits._
+import com.gu.identity.play.IdUser
+import com.gu.support.workers.{Address, BillingPeriod, DigitalPack, User}
+import io.circe.syntax._
+import lib.PlayImplicits._
+import monitoring.SafeLogger
+import monitoring.SafeLogger._
+import play.api.libs.circe.Circe
+import play.api.mvc._
+import services.stepfunctions.{CreateSupportWorkersRequest, StatusResponse, SupportWorkersClient}
+import services.{IdentityService, TestUserService}
+import utils.NormalisedTelephoneNumber
+import utils.NormalisedTelephoneNumber.asFormattedString
+import utils.SimpleValidator._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class CreateSubscription(
+  client: SupportWorkersClient,
+  val actionRefiners: CustomActionBuilders,
+  identityService: IdentityService,
+  testUsers: TestUserService,
+  components: ControllerComponents,
+  settingsProvider: AllSettingsProvider,
+  val supportUrl: String
+)(implicit val ec: ExecutionContext) extends AbstractController(components) with GeoRedirect with CanonicalLinks with Circe with SettingsSurrogateKeySyntax {
+
+  import actionRefiners._
+
+  sealed abstract class CreateSubscriptionError(message: String)
+  case class ServerError(message: String) extends CreateSubscriptionError(message)
+  case class RequestValidationError(message: String) extends CreateSubscriptionError(message)
+
+  type ApiResponseOrError[RES] = EitherT[Future, CreateSubscriptionError, RES]
+
+  def create: Action[CreateSupportWorkersRequest] =
+    authenticatedAction(recurringIdentityClientId).async(circe.json[CreateSupportWorkersRequest]) {
+      implicit request: AuthRequest[CreateSupportWorkersRequest] =>
+        val validator: CreateSupportWorkersRequest => Boolean = request.body.product match {
```

Comment:
This pattern match is obviously pointless at the moment but it gets things ready for adding separate validation for Paper which will be coming soon...

Reply:
I like this idea! But maybe move this stuff into SimpleValidator too? Then the validation stuff can be together and it would be easier to test.

Reply:
Yep good point. I have a follow up PR for Paper specific validation so I will refactor this as part of that change.